packages/features/src/Domain/Feature/Feature.ts

```diff
@@ -0,0 +1,28 @@
+import { PermissionName } from '@standardnotes/auth'
+import { ComponentArea } from '@standardnotes/snjs'
+import { ContentType } from '@standardnotes/snjs'
+import { DockIconType } from './DockIconType'
+import { Flag } from './Flag'
+
+export type Feature = {
+  name: PermissionName;
+  identifier: string;
+  contentType: ContentType
```

Comment:
hmm maybe it make sense to move `ContentType` and `ComponentArea` to `features` package? or extract it to a `content` or `items` package?

The reason for this is that `features` we are not avoiding the circular dependency now:

- when you want to install features, it will install the whole SNJS
- `snjs` will require `features` at some point - and `features` already requires `snjs`
- the server applications will require `features` but not the entire snjs.

wdyt?

Reply:
Ah that makes sense, I tried that option at first but wasn't sure because the whole component model lives on SNJS, so it felt a bit weird to have just ComponentArea live outside of it. Also unit tests don't pass when importing something from an external dependency on a model, but I'm sure we should be able to fix that

Reply:
Done! 🙂 ComponentArea and ContentType now live in features. snjs depends on features and auth, features depends on auth and auth depends on none